salt/transport/tcp.py

```diff
@@ -1385,7 +1388,9 @@ def _publish_daemon(self, **kwargs):
 
         # Spin up the publisher
         pub_server = PubServer(self.opts, io_loop=self.io_loop)
-        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+        sock = socket.socket(socket.AF_INET6 if self.opts['ipv6'] else socket.AF_INET,
+                             socket.SOCK_STREAM)
```

Comment:
Maybe we can also get socket with the sub-function, instead of having the same code thrice.

Reply:
@michaelgibson  Any updates here?